### DISCUSSION TITLE
How to create and connect a Implicit many-to-many relations ?

### DISCUSSION THREAD

**Discussion by Sakkhor909** [prisma/prisma#11838] [Q&A] [answered]
Hello, I am new to Prisma. I have stuck on `Prisma.brand.create` method. Here is the model:

```
model Category {
  id String @id @default(cuid())
  slug String @unique
  name String @unique
  image String?
  createdAt DateTime @default(now())
  user   CMS_users @relation(fields: [userId], references: [id])
  userId String
  brands Brand[]
}

model Brand {
    id String @id @default(cuid())
    slug String @unique
    name String @unique
    image String?
    createdAt DateTime @default(now())
    user   CMS_users @relation(fields: [userId], references: [id])
    userId String
    categories Category[] 
}
```

Basically, I want that when users create a Brand, they choose one or more categories. But I can't figure out how to create a brand with one or more existing connected categories.

How Do I create it on Prisma client? right now it is like this:

```
    prisma.Brand.create({
      data: {
        name: name,
        slug: slug,
        userId: user,
        image: uploadedImage.secure_url,
        categories: {
          create: [
            { category: { connect: { id: "ckzpl0del0014vcv2ctift2f1" } } }
          ]
        }
      },
      include: {
        categories: true
      }
    })
```

My error shows:

```
PrismaClientValidationError: 
Invalid `prisma.brand.create()` invocation:

{
  data: {
    name: 'wassss',
    slug: 'wassss',
    userId: 'd5eea2a7-78d0-41d9-b090-171c9d7100cc',
    image: 'https://res.cloudinary.com/khuji/image/upload/v1645028815/Brands/lfv4cdbhrmwkpzjrz5qc.svg',
    categories: {
    ~~~~~~~~~~
      create: [
        {
          category: {
            connect: {
              id: 'ckzpl0del0014vcv2ctift2f1'
            }
          }
        }
      ]
    }
  },
  include: {
    categories: true
  }
}

Unknown arg `categories` in data.categories for type BrandUncheckedCreateInput. Available args:

type BrandUncheckedCreateInput {
  id?: String
  slug: String
  name: String
  image?: String | Null
  createdAt?: DateTime
  userId: String
}


```

**Comment by nurul3101** [comment, marked as answer]
Hey @Sakkhor909 :wave:,

Just a sanity check, Had you executed `npx prisma generate` after creating your schema? 

I just tried replicating this on my end and it worked as expected.

Are you expecting that while creating a brand the categories would already have been created and they just need to be connected?

In that case this query works as expected

```javascript
  await prisma.brand.create({
    data: {
      name: "Apple",
      slug: "apple",
      categories: {
        connect: {
          id: "ckzr32wlx0000wtt1lrhj85e1",
        },
      },
    },
  });
```
 Please note that the id in categories already exists. I have omitted userId for simplicity purposes.

 Let me know if your use case is different?
 
This is a reference to the [connect guide](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#connect)

**Reply by Sakkhor909** [reply]
Yes the category is already created. I have generate prisma. The database hosted at planate scale. After geting the error, I have also use prisma db push and refresh the scema at planate scale. But still getting the error

**Reply by Sakkhor909** [reply]
I have tried it again but still getting error. the id I am giving is one of the category id copied from prisma studio, still the error is
```
PrismaClientValidationError: 
Invalid `prisma.brand.create()` invocation:

{
  data: {
    name: 'Samsung',
    slug: 'samsung',
    userId: 'd5eea2a7-78d0-41d9-b090-171c9d7100cc',
    image: 'https://res.cloudinary.com/khuji/image/upload/v1645119140/Brands/q9zqaavqhpv4ykmi6nrf.svg',
    categories: {
    ~~~~~~~~~~
      connect: {
        id: 'ckzpkzxdd0005vcv29dhdakfx'
      }
    }
  },
  include: {
    categories: true
  }
}

Unknown arg `categories` in data.categories for type BrandUncheckedCreateInput. Available args:

type BrandUncheckedCreateInput {
  id?: String
  slug: String
  name: String
  image?: String | Null
  createdAt?: DateTime
  userId: String
}
```

Is it a planate scale issue ?

**Reply by nurul3101** [reply]
I am thinking it can have something to do with the fact that PlanetScale doesn't support foreign keys. Let me try to replicate this using PlanetScale.

**Reply by Sakkhor909** [reply]
Is it work in planate scale?

**Comment by Sakkhor909** [comment]
I have read it https://www.prisma.io/docs/concepts/components/prisma-client/relation-queries#connect-an-existing-record everything seems fine. I can't figure out where is the problem.

**Comment by Sakkhor909** [comment]
Note: I didn't promote branch to production in planate scale. I  am using the main branch and there is no other branches

**Reply by Sakkhor909** [reply]
I have also tried with SQLite, Same error 😌

**Comment by danmolloy** [comment]
Some people might find this discussion useful:
https://github.com/prisma/prisma/discussions/4595

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
